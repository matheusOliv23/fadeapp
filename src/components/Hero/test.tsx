import { render, screen } from '@testing-library/react'

import { Hero, HeroProps } from '.'

const data: HeroProps = {
  headlineTitle: 'Construa seu shape sem atrapalhar a sua rotina',
  subtitle:
    'Seu treino personalizado, na palma da sua mão, a qualquer hora e lugar.',
  img: '/background.png',
  textButton: 'EU QUERO',
  minorText: 'Experimente 7 dias Grátis',
}

const { getByRole, getByText, getByTestId } = screen

describe('<Hero />', () => {
  it('should render the hero', () => {
    render(<Hero {...data} />)

    const hero = getByRole('heading', {
      name: 'Construa seu shape sem atrapalhar a sua rotina',
    })
    const subtitle = getByText(
      'Seu treino personalizado, na palma da sua mão, a qualquer hora e lugar.',
    )
    const minorText = getByText('Experimente 7 dias Grátis')
    const image = getByTestId('hero')
    const button = getByRole('button', { name: 'EU QUERO' })
    const { container } = render(<Hero {...data} />)

    expect(hero).toBeInTheDocument()
    expect(subtitle).toHaveTextContent(
      'Seu treino personalizado, na palma da sua mão, a qualquer hora e lugar.',
    )
    expect(minorText).toHaveTextContent('Experimente 7 dias Grátis')
    expect(image).toHaveStyle('backgroundImage: url(/background.png)')
    expect(button).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
