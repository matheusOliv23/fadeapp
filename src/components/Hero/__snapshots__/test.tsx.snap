// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render the hero 1`] = `
<div>
  <div
    class="
  flex max-h-[37.5rem] md:h-[48.25rem] w-full flex-col items-start justify-center p-[2rem] bg-blend-overlay md:p-[9.375rem] object-contain
"
    data-testid="hero"
    style="background-image: url(/background.png); background-position: center; background-size: 100%; background-repeat: no-repeat;"
  >
    <div
      class=" 
flex w-[27.75rem] flex-col items-center justify-center md:items-start md:justify-start 
"
    >
      <h2>
        Construa seu shape sem atrapalhar a sua rotina
      </h2>
      <p
        class="
  mb-5 text-center text-2xl font-normal text-white md:text-left
"
      >
        Seu treino personalizado, na palma da sua mão, a qualquer hora e lugar.
      </p>
      <div
        class="
  flex flex-col last:items-center
"
      >
        <button
          class="flex items-center justify-center rounded-10 bg-primary font-bold text-secondary transition hover:text-primary hover:bg-transparent border-2 border-primary transition-color py-4 w-[21.5rem] text-lg h-[3.625rem] "
        >
          EU QUERO
        </button>
        <p
          class="
  mt-2 text-sm text-gray-500
"
        >
          Experimente 7 dias Grátis
        </p>
      </div>
    </div>
  </div>
</div>
`;
